social/tasks.py

```diff
@@ -0,0 +1,48 @@
+import feedparser
+from background_task import background
+from social.app.models.author import Author
+from social.app.models.post import Post
+
+# Get the GitHub activity of a user
+# Source: https://pypi.python.org/pypi/django-background-tasks
+@background(schedule=60)
+def get_github_activity(authorId):
+    gitAuthor = Author.objects.get(id=authorId)
+    gitUrl = gitAuthor.github
+
+    # lazy way of checking if URL is correct right now
+    if((gitUrl[:19] == "https://github.com/") and (len(gitUrl.split("/")) == 4)):
+        data = feedparser.parse(gitUrl + ".atom")
+
+        # get users post
+        posts = Post.objects.filter(author__id=authorId).filter(title__contains="New GitHub Activity:")
+
+        # Get encoding to decode the data
+        encoding = data["encoding"]
+        # Go over all the entries for the RSS feed, turn them into posts, save
+        # them - this seems to work out okay imo
+        for x in data.get("entries"):
+            found = False
+            gitId = x["id"].encode(encoding).split("/")
+
+            # This is done to avoid adding duplicates
+            for post in posts:
+                entry = post.title.split()
+                if(entry[3] == gitId[1]):
```

Comment:
You're using the title as a unique identifier store. Why not just add a unique field in our `Post` model for the GitHub entry ID? You could then use [`update_or_create`](https://docs.djangoproject.com/en/1.10/ref/models/querysets/#update-or-create) which would improve the performance from O(n^2). Additionally, there would no longer be a need to use a magic number (`entry[3]`) which is highly dependent on the title being "New GitHub Activity:".